### PR TITLE
[SPARK-28184][SQL][TEST] Avoid creating new sessions in SparkMetadataOperationSuite

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

To make the #24972 change smaller. This pr improves `SparkMetadataOperationSuite` to avoid creating new sessions when getSchemas/getTables/getColumns.

## How was this patch tested?

N/A

### CODE REVIEW

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala [hunk 1]
```diff
@@ -17,63 +17,21 @@
...
-
-    def checkResult(dbNames: Seq[String], rs: HiveQueryResultSet): Unit = {
+    def checkResult(rs: ResultSet, dbNames: Seq[String]): Unit = {
       if (dbNames.nonEmpty) {
         for (i <- dbNames.indices) {
           assert(rs.next())
           assert(rs.getString("TABLE_SCHEM") === dbNames(i))
         }
+        // Make sure there are no more elements
+        assert(!rs.next())
```
Comment (srowen) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala]: Just write this once, after the if block?

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala [hunk 2]
```diff
@@ -17,63 +17,21 @@
...
-import org.apache.thrift.protocol.TBinaryProtocol
-import org.apache.thrift.transport.TSocket
+import java.sql.ResultSet
 
 class SparkMetadataOperationSuite extends HiveThriftJdbcTest {
 
   override def mode: ServerMode.Value = ServerMode.binary
 
   test("Spark's own GetSchemasOperation(SparkGetSchemasOperation)") {
-    def testGetSchemasOperation(
```
Comment (srowen) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala]: Help me understand, where does all this deleted logic go? you're saying it just isn't used at all in these tests?

Comment (wangyum) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala]: Before this pr, we constrctor a HiveConnection, a TCLIService.Client, ... and get `ResultSet` from:
```scala
        rs = new HiveQueryResultSet.Builder(connection)
          .setClient(client)
          .setSessionHandle(sessHandle)
          .setStmtHandle(client.GetSchemas(schemaReq).getOperationHandle)
          .build()
```

After this pr, we get `ResultSet` from `statement.getConnection.getMetaData.getSchemas`

cc @juliuszsompolski

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala (hunk 1 above)
Comment (wangyum) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala]: +1

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala [hunk 3]
```diff
@@ -150,75 +68,35 @@ class SparkMetadataOperationSuite extends HiveThriftJdbcTest {
       Seq(
         "CREATE TABLE table1(key INT, val STRING)",
         "CREATE TABLE table2(key INT, val STRING)",
-        "CREATE VIEW view1 AS SELECT * FROM table2").foreach(statement.execute)
+        "CREATE VIEW view1 AS SELECT * FROM table2",
+        "CREATE OR REPLACE GLOBAL TEMPORARY VIEW view_global_temp_1 AS SELECT 1 AS col1",
+        "CREATE OR REPLACE TEMPORARY VIEW view_temp_1 AS SELECT 1 as col1"
```
Comment (juliuszsompolski) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala]: Handling of views depends on #24972, but you want to add it now, and then change the answers in #24972 ? Fine by me.

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala (hunk 1 above)
Comment (juliuszsompolski) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala]: +1

**Review by juliuszsompolski — APPROVED** [review]
LGTM pending nit from @srowen

**Review by juliuszsompolski — APPROVED** [review]

**Review by gatorsmile — COMMENTED** [review]
LGTM

Thanks! Merged to master.

**Pull request closed without merging**